Sources/Components/Modal/Modal.swift

```diff
@@ -0,0 +1,237 @@
+        private let colorProvider: ColorProvider
+        
+        /**
+         - Parameter title: The main title text to be displayed.
+         - Parameter subtitle: An optional subtitle text to provide additional context or information. Defaults to `nil`.
+         - Parameter bodyText: The main body text content of the component.
+         - Parameter primaryButton: An optional primary button constructor for defining the primary action. Defaults to `nil`.
+         - Parameter secondaryButton: An optional secondary button constructor for defining a secondary action. Defaults to `nil`.
+         - Parameter closeButton: A Boolean value indicating whether a close button should be shown. Defaults to `false`.
+         - Parameter isPresented: A binding to a Boolean value that controls the visibility of the component.
+         - Parameter colorProvider: A provider for the color scheme of the component. Defaults to `Config.colorProvider`.
+         */
+        public init(
+            title: String,
+            subtitle: String? = nil,
+            bodyText: String,
+            primaryButton: ButtonConstructor? = nil,
+            secondaryButton: ButtonConstructor? = nil,
+            closeButton: Bool = false,
+            isPresented: Binding<Bool>,
+            colorProvider: ColorProvider = Config.colorProvider
+        ) {
+            self.title = title
+            self.subtitle = subtitle
+            self.bodyText = bodyText
+            self.primaryButtonProvider = primaryButton
+            self.secondaryButtonProvider = secondaryButton
+            self.closeButton = closeButton
+            self._isPresented = isPresented
+            self.colorProvider = colorProvider
+        }
+        
+        public var body: some View {
+            ZStack {
+                backgroundView
+
+                foregroundView
+            }
+            .fixedSize(horizontal: false, vertical: true)
+            .cornerRadius(modalCornerRadius)
+            .shadow(.small)
+            .accessibilityElement(children: .combine)
+            .padding()
+        }
+        private var titleView: some View {
+            Text(title, style: .title3)
+                .accessibilityAddTraits(.isHeader)
+        }
+        
+        @ViewBuilder
+        private var closeButtonView: some View {
+            if closeButton {
+                Image("icon-close", bundle: .module)
+                    .onTapGesture {
+                        isPresented = false
+                    }
+            }
+        }
+        
+        @ViewBuilder
+        private var subtitleView: some View {
+            if let subtitle, !subtitle.isEmpty {
+                Text(subtitle, style: .title4)
+                    .accessibilityRemoveTraits(.isHeader)
+            }
+        }
+        
+        private var bodyTextView: some View {
+            Text(bodyText, style: .body)
+                .accessibilityRemoveTraits(.isHeader)
```

Comment:
```suggestion
            Text(bodyText, style: .body)
                .fixedSize(horizontal: false, vertical: true)
                .accessibilityRemoveTraits(.isHeader)
```

Reply:
I noticed that the modal clips text on an iPhone 13 mini as you can see here:

![Simulator Screenshot - iPhone 13 mini - 2024-06-25 at 11 19 48](https://github.com/warp-ds/warp-ios/assets/117636/feda8d39-9e39-4d84-872a-23be7b713e51)

I guess the error is similar to the one I've attempted to fix in #44 where the solution was to introduce a `.fixedSize` modifier like I've suggested above.

You should probably check if that introduces any side effects elsewhere but I just wanted to suggest a - potential - fix instead of just dumping an error on you 😄

Reply:
Yes, `.fixedSize(horizontal: false, vertical: true)` seems fix the issue.